src/crimson/mon/MonClient.cc

```diff
@@ -955,7 +967,8 @@ seastar::future<> Client::reopen_session(int rank)
       }
     }).then([peer, this](auto result) {
       if (result == Connection::AuthResult::canceled) {
-	return seastar::now();
+	return seastar::make_exception_future(
```

Comment:
If we return a valid future(), it would trigger an assert at [here](https://github.com/ceph/ceph/blob/6a855bf23a4ead146f0cc0539a040968cbd1a2ba/src/crimson/mon/MonClient.cc#L998)

Reply:
i think the reason why i think we should avoid throwing or returning an exception here is that `seastar::parallel_for_each()` should not return a value with exception in it if *any* of of the pending connections to monitor is authenticated and promoted to "active" connection.

by translating "Connection::AuthResult::canceled" to a "network_aborted" and fail the call is far from what we expect here.